File: backend/src/modules/accounts/entities/account.entity.ts
export class AccountEntity {
  id!: string;
  userId!: string;
  name!: string;
  type!: string; // BANK, CREDIT_CARD, DIGITAL_WALLET, CASH, UPI, INVESTMENT
  balance!: number;
  currency!: string;
  bankName?: string | null;
  accountNumber?: string | null;
  routingNumber?: string | null;
  isDefault!: boolean;
  color?: string | null;
  icon?: string | null;
  createdAt!: Date;
  updatedAt!: Date;
  deletedAt?: Date | null;

  constructor(partial: Partial<AccountEntity>) {
    Object.assign(this, partial);
  }
}

File: mobile/lib/features/accounts/domain/entities/account_entity.dart
import 'package:equatable/equatable.dart';

class AccountEntity extends Equatable {
  final String id;
  final String userId;
  final String name;
  final String type; // BANK, CREDIT_CARD, DIGITAL_WALLET, CASH, UPI, INVESTMENT
  final double balance;
  final String currency;
  final String? bankName;
  final String? accountNumber;
  final String? routingNumber;
  final bool isDefault;
  final String? color;
  final String? icon;
  final DateTime createdAt;
  final DateTime updatedAt;

  const AccountEntity({
    required this.id,
    required this.userId,
    required this.name,
    required this.type,
    required this.balance,
    required this.currency,
    this.bankName,
    this.accountNumber,
    this.routingNumber,
    required this.isDefault,
    this.color,
    this.icon,
    required this.createdAt,
    required this.updatedAt,
  });

  @override
  List<Object?> get props => [
        id,
        userId,
        name,
        type,
        balance,
        currency,
        bankName,
        accountNumber,
        routingNumber,
        isDefault,
        color,
        icon,
        createdAt,
        updatedAt,
      ];
}
